include/oc_cloud.h

```diff
@@ -71,6 +71,7 @@ typedef enum {
   CLOUD_ERROR_RESPONSE = 1,
   CLOUD_ERROR_CONNECT = 2,
   CLOUD_ERROR_REFRESH_ACCESS_TOKEN = 3,
+  CLOUD_ERROR_UNAUTHORIZED = 4,
```

Comment:
do we need a new error? why you introduce it?

Reply:
I check it in `cloud_ep_session_event_handler`:

```
static void
cloud_ep_session_event_handler(const oc_endpoint_t *endpoint,
                               oc_session_state_t state)
{
  oc_cloud_context_t *ctx = oc_cloud_get_context(endpoint->device);
  if (ctx && oc_endpoint_compare(endpoint, ctx->cloud_ep) == 0) {
    OC_DBG("[CM] cloud_ep_session_event_handler ep_state: %d\n", (int)state);
    ctx->cloud_ep_state = state;
    if (ctx->cloud_ep_state == OC_SESSION_DISCONNECTED &&
      ctx->cloud_manager &&
      !(ctx->store.cps == OC_CPS_FAILED && ctx->last_error == CLOUD_ERROR_UNAUTHORIZED)) {
      cloud_manager_restart(ctx);
    }
  }
}
```

There are multiple parts of code that set cps to `OC_CPS_FAILED`, but I haven't examined it too deeply, maybe the condition can be simplified to:

```
...
    if (ctx->cloud_ep_state == OC_SESSION_DISCONNECTED &&
      ctx->cloud_manager &&
      ctx->store.cps != OC_CPS_FAILED)
...
```
Though I'll have to check that (maybe documenting somewhere when OC_CPS_FAILED is set would be useful anyway). Or do you have some other suggestion? Other possibility would be to call `oc_cloud_manager_stop` which removes `cloud_ep_session_event_handler` from session callbacks. Though I don't know whether that's desired or not.

Reply:
new error state removed